codeql-analysis.yml:initial creation

andy5995 opened the pull request.

andy5995 committed:
codeql-analysis.yml:initial creation

andy5995 commented:
Do you want this added?

Either the [errors it found](https://github.com/widelands/widelands/pull/5342/checks?check_run_id=5778976244) could be fixed in a separate PR, and afterward this branch could be rebased before merging.

Or maybe it would be better to do the fixes on this PR so codeql could test the changes before merging them.

Unless you're not worried about type conversion errors ;) Then just close the PR :)

Noordfrees reviewed (commented):
+1

Since the hits are all about narrow-type-wide-type mismatches it would IMHO be best to fix them all in this branch.

---

Since you're contributing quite a lot to Widelands, please also add yourself to the `data/txts/developers.json` :)

andy5995 commented:
> Since you're contributing quite a lot to Widelands, please also add yourself to the `data/txts/developers.json` :)

Thanks!

andy5995 commented:
For anyone interested, docs for running codeql from the command line are at https://codeql.github.com/docs/codeql-cli/creating-codeql-databases/

andy5995 committed:
remove paths-ignore field

hessenfarmer reviewed (changes requested):
some comments inline 
thios should be checked carefully by @Noordfrees as some cha ge slook dangerous in my eyes

hessenfarmer commented:
in the test suite the scan results of codeql is finished after 2 seconds while the check is still running. maybe there is a depends missing.

andy5995 commented:
> in the test suite the scan results of codeql is finished after 2 seconds while the check is still running. maybe there is a depends missing.

I'm a bit confused by that too. shortly after I push, I see

![image](https://user-images.githubusercontent.com/16764864/161886290-730190bb-8ae7-4c87-bba9-23b98e5a31f8.png)

But there's also this:

![image](https://user-images.githubusercontent.com/16764864/161886368-624cdf26-6d59-48a9-b87a-bdcc06fe6efa.png)

andy5995 commented:
I've opened a PR on my fork that adds the codeql analysis file but doesn't introduce any fixes. The results should be at https://github.com/andy5995/widelands/pull/2/checks?check_run_id=5844248996

It should soon show all original 52 errors codeql complained about.

andy5995 committed:
run build manually so multiple cores can be specified

andy5995 commented:
Tests now failing. Why is this happening? https://github.com/widelands/widelands/runs/5844695311?check_suite_focus=true#step:5:1305

andy5995 committed:
fix cmake option (build translations off)

andy5995 committed:
cast players.size() as uint8_t instead of Widelands::PlayerNumber

andy5995 committed:
defaultai.cc:L5000, remove (probably) useless cast

andy5995 committed:
codeql-analysis.yml:change to build directory before compiling

andy5995 committed:
spritesheet_animation.cc:L134/revert changes, upcast 'rows'

to fix "Multiplication result converted to larger type" (codeql)

https://github.com/andy5995/widelands/security/code-scanning/2

andy5995 committed:
fields_to_draw.cc:L114/revert changes, upcast '_w'

Fix for https://github.com/andy5995/widelands/security/code-scanning/3

And thanks to @hessenfarmer :)

andy5995 committed:
'src/ai/defaultai.cc' was automatically formatted.

andy5995 committed:
codeql-analysis.yml:turn off ASAN

andy5995 committed:
codeql-analysis.yml:revert to autobuild

Python test wasn't being run

andy5995 committed:
declare some descriptions as uint16_t

andy5995 committed:
'src/logic/map_objects/tribes/tribe_descr.cc' was automatically formatted.

andy5995 commented:
rebased at https://github.com/widelands/widelands/commit/b2addfc83a51e1f0d27a5a33831e8a439a077b20

andy5995 committed:
.github/workflows/codeql-analysis.yml:install deps with existing script

andy5995 committed:
declare vars for loops using DescriptionIndex type

andy5995 committed:
.github/workflows/codeql-analysis.yml:add concurrency cancel rule

hessenfarmer commented:
@andy5995 do you still want to work on this.
@Noordfrees from my perspective this could be postponed to 1.2 as it is no0t a feature but another codechecking / codeimprovement fix

andy5995 commented:
> @andy5995 do you still want to work on this.

At this time, no.

hessenfarmer commented:
@andy5995 @Noordfrees 
maybe we should skip this and enable the relating checks of the majorities of results of codeql on our code. 
Namely: 
bugprone-implicit-widening-of-multiplication-result
bugprone-too-small-loop-variable

andy5995 commented:
Sure, I'm not emotionally attached to this PR.
